charts/spv-wallet-stack/charts/spv-wallet/templates/deployment.yml

```diff
@@ -65,16 +65,11 @@ spec:
                   name: {{ .Release.Name }}-postgresql-spv-wallet
                   key: password
           {{- end }}
-          {{- if .Values.config.paymail.beef.block_headers_service_auth_token }}
-            - name: SPVWALLET_PAYMAIL_BEEF_BLOCK_HEADERS_SERVICE_AUTH_TOKEN
-              value: {{ .Values.config.paymail.beef.block_headers_service_auth_token }}
-          {{- else}}
             - name: SPVWALLET_PAYMAIL_BEEF_BLOCK_HEADERS_SERVICE_AUTH_TOKEN
               valueFrom:
                 secretKeyRef:
                   name: block-headers-service-secret
```

Comment:
the name should be different depending on that if the value .Values.config.paymail.beef.block_headers_service_auth_token is set.
1. if it is not set, then name should be `name: block-headers-service-secret`
2. if it is set then name should be `name: {{ include "spv-wallet.fullname" . }}`

HINT, you can do this with either if statement or ternary function, something like:
```
{{- $bhsToken := .Values.config.paymail.beef.block_headers_service_auth_token -}}
name: {{- $bhsToken | ternary "block-headers-service-secret" {{ include "spv-wallet.fullname" . }} }}
```

Reply:
Now I understand. Corrections added.